trl/trainer/ppo_trainer.py

```diff
@@ -625,10 +625,9 @@ def collator(data):
                     vpreds,
                     batch["masks"],
                 )
-                if self.config.early_stopping and train_stats["policy/policykl"] > 1.5 * self.config.target_kl:
-                    early_stop = True
-                    self.optimizer.zero_grad()
-                    break
+                if self.config.early_stopping:
+                    policykl = train_stats["policy/policykl"]
+                    early_stop = self._early_stop(policykl)
```

Comment:
Do we not want to break here if early stop is True? Otherwise we might remain in the inner loop for a few more iterations.

Reply:
Agreed! this should be fixed now